Lib/unittest/runner.py

```diff
@@ -45,7 +45,8 @@ def __init__(self, stream, descriptions, verbosity, *, durations=None):
         self.showAll = verbosity > 1
         self.dots = verbosity == 1
         self.descriptions = descriptions
-        self._ansi = get_colors()
+        file = sys.stderr if stream == "<stderr>" else None
```

Comment:
???

`stream` is normally an instance of `_WritelnDecorator`.

Reply:
This was a typo, I meant to do:

```suggestion
        file = sys.stderr if stream.name == "<stderr>" else None
```

But perhaps we should do something like this?

```python
        colorize = False
        if hasattr(stream, "name"):
            if stream.name == "<stdout>":
                colorize = can_colorize(file=sys.stdout)
            elif stream.name == "<stderr>":
                colorize = can_colorize(file=sys.stderr)
        self._ansi = get_colors(colorize=colorize)
```

Same for below.

Reply:
Why not simply use `file=stream`?